### DISCUSSION TITLE
Terrible generations

### DISCUSSION THREAD

**Discussion by VenomousValentine** [lshqqytiger/stable-diffusion-webui-directml#187] [General]
Has anyone noticed their generations looking absolutely terrible since the latest update? Bad anatomy, jank faces, just lots of inconsistency.

**Comment by pw405** [comment]
Yes... and I have been tinkering with command line arguments trying to figure out what is happening.  

I've even had results like this, but when using `--precision autocast` in the batch file.  HOWEVER, I determined this was the result of using textual inversions "Bad Dream" and "Unrealistic Dream"  (from Lykon, creator of popular Dreamshapers model).  Without the textual inversions, it would generate successfully.  (Using a 7900XTX). 

It isn't completely broken, but something just seems "off" lately. I haven't got to test much.  

This is generally safe to use for me: 
`--medvram --opt-sdp-attention --no-half --precision full`

Supposed to be Aragorn from Lord of the Rings :
![00005-3203515382](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/20309609/c4faa096-4932-4fd2-99ea-84c879e08f6b)

**Reply by EygenCat** [reply]
Thank you so much, I couldn't understand what was wrong with Textual Inversion. When using Textual Inversion, everything was breaking, but without Textual Inversion, everything was working fine. And after I added the command line arguments 
`--medvram --opt-sdp-attention --no-half --precision full`, 
everything became great.

**Reply by EygenCat** [reply]
But my Hires.fix function stopped working. I removed a few arguments, and now my arguments look like this:
`set COMMANDLINE_ARGS=--opt-sub-quad-attention --disable-nan-check --precision autocast --autolaunch --listen --no-half --precision full`
it seems like it's generating images perfectly fine.

**Reply by pw405** [reply]
Oh good!  Note, the --opt-sub-quad-attention can be set in the Web UI options in the optimization section if you don't want to have it in the command line.  I started selecting sdp attention there instead of command line.

**Comment by EygenCat** [comment]
Unfortunately, I couldn't make Hires.fix work properly. It only works intermittently and most of the time it throws a 
`RuntimeError: Could not allocate tensor with 1207959552 bytes. There is not enough GPU video memory available! error.`
I have rx 6700 xt

Apparently I will be sitting on the old version

**Reply by pw405** [reply]
See, I've never used HiRes fix. Is this any different than simply upscaling using the Extras section?

Another method I use for Upscaling is the Control Net Tile Resample along with "Ultimate SD Upscale" script.  It re-samples an image using  ~15 512x512 images.  Excellent results, and supposed to be easier on VRAM. May be worth checking out!

**Reply by EygenCat** [reply]
"Hires.fix" is needed to generate images in a higher resolution without distorting the composition. Essentially, it upscales the image using one of the provided methods and then performs img2img processing.

**Comment by VenomousValentine** [comment]
None of any of this helped. I even tried reinstalling, and my generations are still ugly as sin despite using the same prompts I used to.

**Reply by fishonainternet** [reply]
I'm currently in big regret after updating my Stable Diffusion to the newest version (1.4.1) haha, is there a way to revert back to old version?

**Reply by pw405** [reply]
What GPU are you using?  This seems really strange that the new version created so many issues.

**Reply by VenomousValentine** [reply]
6600 xt

**Reply by Siralob** [reply]
I had same problem. Everything generated was like above post. I had to reset my webui because I re-install my `venv` after I upgraded latest to try to solve this issue. I thought it's broken for me. I also tried switched to spd, other optimization... and webui said no to me.

I reset back from latest to Commit HASH:[265d626471eacd617321bdb51e50e4b87a7ca82e] and work again.
~~But, I still got plane image or black image while generating with Doggtexx randomly. So, I have no idea. Also, some error code like `mat1 and mat2...` something like that.(sorry I forgot to backup them)~~

Which commit hash you used before this upgrade? Maybe, you could find your commit hash you used before on [here](https://github.com/lshqqytiger/stable-diffusion-webui-directml/commits/master).

Before the reinstall or `git reset --hard [CommitHASH]` just put your desire HASH no need [ ].
I revert back to old version, then re-install my `venv`, too.

Before you reset, these files will be replaced to original. 
```
webui-user.bat
config.json
ui-config.json
params.txt
styles.csv
user.css
```

my webui-user.bat settings
```
set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128
set COMMANDLINE_ARGS= --medvram --precision full --no-half --opt-split-attention --disable-nan-check --api --autolaunch
set SAFETENSORS_FAST_GPU=1
```

My question, Does anyone know which commit hash is the most suitable and stable for 8GB card like RX400 and RX500?(e.g. RX580 8GB) Also, RDNA2 card can be different for stable version. 
I have no luck with hires.fix nor above 768x768 resolution for OOM (sometimes works with 512x912, 512x760).

**Comment by EygenCat** [comment]
Version 1 5 1 works great!!!

**Comment by VenomousValentine** [comment]
I tried the latest update. its still terrible. Still haven't been able to install previous versions that work.

**Reply by pw405** [reply]
What command line arguments are you using?

**Reply by VenomousValentine** [reply]
I've tried a bunch. they don't change anything.
